System/Atom.h

```diff
@@ -0,0 +1,558 @@
+		/// </returns>
+		virtual int ReadProperty(std::string propName, Reader &reader);
+
+		/// <summary>
+		/// Saves the complete state of this Atom to an output stream for later recreation with Create(Reader &reader).
+		/// </summary>
+		/// <param name="writer">A Writer that the Atom will save itself with.</param>
+		/// <returns>An error return value signaling success or any particular failure. Anything below 0 is an error signal.</returns>
+		virtual int Save(Writer &writer) const;
+#pragma endregion
+
+#pragma region Memory Management
+		/// <summary>
+		/// Grabs from the pre-allocated pool, an available chunk of memory the exact size of an Atom. OWNERSHIP IS TRANSFERRED!
+		/// </summary>
+		/// <returns>A pointer to the pre-allocated pool memory. OWNERSHIP IS TRANSFERRED!</returns>
+		static void * GetPoolMemory();
+
+		/// <summary>
+		/// Adds a certain number of newly allocated instances to this' pool.
+		/// </summary>
+		/// <param name="fillAmount">The number of instances to fill with. If 0 is specified, the set refill amount will be used.</param>
+		static void FillPool(int fillAmount = 0);
+
+		/// <summary>
+		/// Returns a raw chunk of memory back to the pre-allocated available pool.
+		/// </summary>
+		/// <param name="returnedMemory">The raw chunk of memory that is being returned. Needs to be the same size as an Atom. OWNERSHIP IS TRANSFERRED!</param>
+		/// <returns>The count of outstanding memory chunks after this was returned.</returns>
+		static int ReturnPoolMemory(void *returnedMemory);
+#pragma endregion
+
+#pragma region Getters and Setters
+		/// <summary>
+		/// Gets the current owner MovableObject of this AtomGroup.
+		/// </summary>
+		/// <returns>A const pointer to the owner.</returns>
+		const MovableObject * GetOwner() const { return m_OwnerMO; }
+
+		/// <summary>
+		/// Sets the current owner MovableObject of this AtomGroup.
+		/// </summary>
+		/// <param name="newOwner">A pointer to the new owner. Ownership is NOT transferred!</param>
+		void SetOwner(MovableObject *newOwner) { m_OwnerMO = newOwner; }
+
+		/// <summary>
+		/// Gets the group ID of this Atom.
+		/// </summary>
+		/// <returns>The group ID of this Atom.</returns>
+		long GetSubID() const { return m_SubgroupID; }
+
+		/// <summary>
+		/// Sets the subgroup ID of this Atom.
+		/// </summary>
+		/// <param name="newID">The new subgroup ID of this Atom.</param>
+		void SetSubID(long newID = 0) { m_SubgroupID = newID; }
+
+		/// <summary>
+		/// Gets the material of this Atom.
+		/// </summary>
+		/// <returns>The material of this Atom.</returns>
+		Material const * GetMaterial() const { return m_Material ? m_Material : g_SceneMan.GetMaterialFromID(g_MaterialAir); }
+
+		/// <summary>
+		/// Sets the material of this Atom.
+		/// </summary>
+		/// <param name="newMat">The new material of this Atom.</param>
+		void SetMaterial(Material *newMat) { m_Material = newMat; }
+
+		/// <summary>
+		/// Gets the Color of this Atom's trail.
```

Comment:
Yeah MOPixels use trails, cause they're basically rebranded Atoms.
It would make sense to pull the handling out and move it over to them but that's probably not gonna be simple.

Reply:
Okay it's for MOPixel trails? I wasn't sure if that's where those trails were coming from or if this was some old thing that's no longer used. As you say, probably worth pulling them out but also fuck that noise.